src/pages/AdminPage.jsx

```diff
@@ -0,0 +1,213 @@
+import { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchProducts, addProduct, deleteProduct } from '../redux/products/productSlice';
+import './AdminPage.css';
+
+const AdminProductManagement = () => {
+  const dispatch = useDispatch();
+  const [productName, setProductName] = useState('');
+  const [productPrice, setProductPrice] = useState('');
+  const [productPicture, setProductPicture] = useState('');
+  const [productCPU, setProductCPU] = useState('');
+  const [productMemory, setProductMemory] = useState('');
+  const [productStorage, setProductStorage] = useState('');
+  const [searchKeyword, setSearchKeyword] = useState('');
+  const { products, isLoading, isError } = useSelector((state) => state.products);
+
+  useEffect(() => {
+    dispatch(fetchProducts());
+  }, [dispatch]);
+
+  const [formErrors, setFormErrors] = useState({
+    name: '',
+    price: '',
+    picture: '',
+    cpu: '',
+    memory: '',
+    storage: '',
+  });
+
+  const validateForm = () => {
+    let isValid = true;
+    const errors = {};
+
+    if (!productName.trim()) {
+      isValid = false;
+      errors.name = 'Product name is required';
+    }
+    if (!productPicture.trim()) {
+      isValid = false;
+      errors.picture = 'Product picture is required';
+    }
+    if (!productCPU.trim()) {
+      isValid = false;
+      errors.cpu = 'Product cpu is required';
+    }
+    if (!productMemory.trim()) {
+      isValid = false;
+      errors.memory = 'Product memory is required';
+    }
+    if (!productStorage.trim()) {
+      isValid = false;
+      errors.storage = 'Product storage is required';
+    }
+
+
+    if (!productPrice.trim()) {
+      isValid = false;
+      errors.price = 'Product price is required';
+    } else if (isNaN(productPrice) || +productPrice <= 0) {
+      isValid = false;
+      errors.price = 'Product price must be a positive number';
+    }
```

Comment:
I would suggest that instead of manually checking the presence of every value, just use the `required` attribute for the corresponding inputs in the form.

Reply:
We try this for positive number validation